pkg/kallsyms/kallsyms_test.go

```diff
@@ -198,29 +237,147 @@ func TestRealKAllSymsParsing(t *testing.T) {
-// reading the kallsyms file
-func BenchmarkKernelSymbolAddressJITParsing(b *testing.B) {
-	utilstest.RequireRoot(b)
-	b.ReportAllocs()
+func TestNewKAllSymsFromReaderErrors(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       string
+		expectedErr string
+	}{
+		{
+			name:        "invalid_format_less_fields",
+			input:       "0000000000000000 A",
+			expectedErr: "line \"0000000000000000 A\" has less than 3 fields",
+		},
+		{
+			name:        "invalid_address",
+			input:       "INVALID_ADDR A fixed_percpu_data",
+			expectedErr: "strconv.ParseUint",
+		},
+		{
+			name:        "malformed_line_no_fields",
+			input:       "0000000000000000",
+			expectedErr: "line \"0000000000000000\" has less than 3 fields",
+		},
+	}
 
-	for n := 0; n < b.N; n++ {
-		exists := SymbolExists("security_bprm_check")
-		require.Equal(b, true, exists, "exists")
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := strings.NewReader(tt.input)
+			_, err := NewKAllSymsFromReader(reader)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), tt.expectedErr)
+		})
 	}
 }
 
-// BenchmarkKernelSymbolAddressPreLoading loads the full kallsyms before
-// looking up the symbol
-func BenchmarkKernelSymbolAddressPreLoading(b *testing.B) {
-	utilstest.RequireRoot(b)
-	b.ReportAllocs()
-
-	for n := 0; n < b.N; n++ {
-		k, err := NewKAllSyms()
-		require.NoError(b, err)
-		require.NotNil(b, k)
-		_, exists := k.symbolsMap["security_bprm_check"]
-		require.Equal(b, true, exists)
+func TestSymbolExistsWithEmptySymbol(t *testing.T) {
+	exists := SymbolExists("")
+	require.False(t, exists)
+}
+
+// MockVar represents a mock implementation of ebpf.Var for testing
+type MockVar struct {
+	value interface{}
+}
+
+func (v *MockVar) Set(value interface{}) error {
+	v.value = value
+	return nil
+}
+
+// mockSymbolResolver implements symbolResolver interface for testing
+type mockSymbolResolver struct {
+	resolveFunc func(symbol string) (uint64, error)
+}
+
+func (m *mockSymbolResolver) resolve(symbol string) (uint64, error) {
+	return m.resolveFunc(symbol)
+}
+
+func TestSpecUpdateAddressesEmptySymbols(t *testing.T) {
+	spec := &ebpf.CollectionSpec{
+		Maps:      make(map[string]*ebpf.MapSpec),
+		Programs:  make(map[string]*ebpf.ProgramSpec),
+		Variables: make(map[string]*ebpf.VariableSpec),
 	}
+	err := SpecUpdateAddresses(spec, []string{})
+	require.NoError(t, err)
+}
+
+func TestSpecUpdateAddressesSymbolResolution(t *testing.T) {
+	// Reset global state before each test
+	symbolsMap = make(map[string]uint64)
+	triedGetAddr = make(map[string]error)
+
+	t.Run("all_resolvers_fail", func(t *testing.T) {
+		resolver1 := &mockSymbolResolver{
+			resolveFunc: func(symbol string) (uint64, error) {
+				return 0, fmt.Errorf("first resolver failed")
+			},
+		}
+		resolver2 := &mockSymbolResolver{
+			resolveFunc: func(symbol string) (uint64, error) {
+				return 0, fmt.Errorf("second resolver failed")
```

Comment:
This error message could be defined inside the `t.Run("all_resolvers_fail", func(t *testing.T) {` function as a const. Then the variable can be used in the `.Contains` check below